mesh/ncmesh.cpp

```diff
@@ -5504,6 +5504,30 @@ void NCMesh::LoadVertexParents(std::istream &input)
    }
 }
 
+void NCMesh::LoadVertexToKnot(std::istream &input)
+{
+   int nv;
+   input >> nv;
+   MFEM_VERIFY(0 <= nv, "Invalid vertex-to-knot data");
+   vertex_to_knot.SetSize(4 * nv);
+   for (int i=0; i<nv; ++i)
+   {
+      int id, p1, p2, k;
+      input >> id >> k >> p1 >> p2;
+      MFEM_VERIFY(input, "problem reading vertex parents.");
+
+      MFEM_VERIFY(nodes.IdExists(id), "vertex " << id << " not found.");
+      MFEM_VERIFY(nodes.IdExists(p1), "parent " << p1 << " not found.");
+      MFEM_VERIFY(nodes.IdExists(p2), "parent " << p2 << " not found.");
+      MFEM_VERIFY(0 < k, "Invalid knot index");
+
+      vertex_to_knot[(4*i) + 0] = id;
```

Comment:
Instead of loading 4 consecutive `int`s perhaps a small `struct` could be considered.

Advantage could be transparency: what does the data represent. Might also be more future proof in case more data needs to be added per vertex.

Downside perhaps I am over thinking things and slowing execution down.